server/forge/common/utils.go

```diff
@@ -37,3 +43,20 @@ func ExtractHostFromCloneURL(cloneURL string) (string, error) {
 
 	return host, nil
 }
+
+func UserToken(ctx context.Context, r *model.Repo, u *model.User) string {
+	if u != nil {
+		return u.Token
+	}
+
+	_store, ok := store.TryFromContext(ctx)
+	if !ok {
+		log.Error().Msg("could not get store from context")
+		return ""
+	}
+	user, err := _store.GetUser(r.UserID)
```

Comment:
hmm we might als add a nil check for r

Reply:
I added a simple check that in best case will never be exec, in worst case we have a unique error msg we can trace down